docs/ccv1_image_security_design.md

```diff
@@ -0,0 +1,403 @@
+
+3. Use the public key to verify the cryptographic signature in the signature claim file, and then compare whether the manifest digest in the JSON payload is consistent with that of the container image and whether the docker reference in the JSON payload is consistent with that of the container image.
+
+### Image decryption
+
+After verifying the container image signature, will start the actual image pulling. When pulling the image layer by layer, if encounter an encrypted image layer, image-rs calls the container image cryptography operation library provided by ocicrypt-rs to decrypt the encrypted image layer as follows:
+
+1. Read the layer's two annotations fields in the image manifest. The `org.opencontainers.image.enc.keys.provider.attestation` field is the annotation packet mentioned above, which contains `PrivateLayerBlockCipherOptions` encrypted by the user's private key. The symmetric key used to encrypt this layer is recorded in this `PrivateLayerBlockCipherOptions`.
+2. ocicrypt-rs calls the keyprovider grpc service provided by the attestation-agent and sends the annotation packet to the attestation-agent.
+3. The attestation-agent performs HW-TEE remote attestation with the relying KBS deployed by the remote user. After passing the attestation, it interacts with KBS through the encrypted channel, and decrypts the `PrivateLayerBlockCipherOptions` using the user's public key in the KBS.
+4. ocicrypt-rs obtains the decrypted `PrivateLayerBlockCipherOptions` from the attestation-agent, reads the symmetric key used to decrypt the container image, and completes the decryption of the container image layer.
+
+## Attestation Agent
+
+The attestation-agent is an indispensable core component in the confidential containers architecture. It undertakes the trust distribution function of the confidential container. In the process of signature verification and decryption of the protected image, the attestation-agent serves as the source of the user's confidential information and the decryptor of the `PrivateLayerBlockCipherOptions`. As mentioned above, it provides two gRPC services for image-rs and ocicrypt-rs to support the above two functions.
+
+### GetResource service
+
+The GetResource grpc service interface provided by the attestation agent is used to obtain various confidential resources from the relying party (KBS). In the process of protected image deployment, it is usually used to obtain policiy.json and sigstore config file. 
+
+Attestation-agent performs HW-TEE attestation to the KBS, establish an encrypted channel after negotiating the key, and download the required confidential resources through the encrypted channel. The GetResource grpc service interface protobuf of the attachment agent is defined as follows:
+
+```protobuf
+message getResourceRequest {
+    string KbcName = 1;
+    string KbsUri = 2;
+    string ResourceDescription = 3;
```

Comment:
What is `ResourceDescription` used for? In the examples below, it seems mostly empty.

Reply:
Yes, because the service interface of KBS is not standardized, and the communication protocol between KBC and KBS may have high customization requirements, in order to leave enough flexibility for the service implementation of KBS, I think we should separate the resource name and information from the address of KBS